pkg/runtime/runtime.go

```diff
@@ -2347,6 +2347,12 @@ func (a *DaprRuntime) convertMetadataItemsToProperties(items []components_v1alph
 		for strings.Contains(val, "{uuid}") {
 			val = strings.Replace(val, "{uuid}", uuid.New().String(), 1)
 		}
+		for strings.Contains(val, "{podName}") {
+			if a.podName == "" {
+				log.Warnf("could not get podName property")
```

Comment:
Please, fail instead. Component should not init at all and give clear error. Warnings can easily be ignored.

Reply:
Ping @CacheCoww please fix this so we can merge.